File: 10/03/pages/dynamic_loading_page.py
from selenium.common.exceptions import TimeoutException
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait


class DynamicLoadingPage():
    _start_button = {"by": By.CSS_SELECTOR, "value": "#start button"}
    _finish_text = {"by": By.ID, "value": "finish"}

    def __init__(self, driver):
        self.driver = driver

    def load_example(self, example_number):
        self.driver.get("http://the-internet.herokuapp.com/dynamic_loading/" + example_number)
        self.driver.find_element(self._start_button["by"], self._start_button["value"]).click()

    def finish_text_present(self):
        return self._wait_for_is_displayed(self._finish_text, 10)

    def _wait_for_is_displayed(self, locator, timeout):
        try:
            wait = WebDriverWait(self.driver, timeout)
            wait.until(
                expected_conditions.visibility_of_element_located(
                    (locator['by'], locator['value'])))
        except TimeoutException:
            return False
        return True

File: code-examples/javascript/11/03/pages/DynamicLoadingPage.js
const Page = require('./Page')

const START_BUTTON = { css: '#start button' }
const FINISH_TEXT = { id: 'finish' }

class DynamicLoadingPage extends Page {
  constructor(driver) {
    super(driver)
  }

  async loadExample(exampleNumber) {
    await visit(
      'http://the-internet.herokuapp.com/dynamic_loading/' + exampleNumber
    )
    await click(START_BUTTON)
  }

  async isFinishTextPresent() {
    return isDisplayed(FINISH_TEXT, 10000)
  }
}

module.exports = DynamicLoadingPage
